### ISSUE TITLE
[FIX] Python SLM accepts negative numbers in tri

### ISSUE THREAD

**Issue by ReinderVosDeWael** [MICA-MNI/BrainStat#163]
The Python SLM accepts negative numbers in its tri field. This doesn't result in downstream errors because indexing by negative numbers is allowed in Python. 

Proposed fix: Check for negative numbers whenever tri is set.

**Comment by RandallWert** [comment]
A few questions:

1. Would this bug fix involve only the file BrainStat/brainstat/stats/SLM.py?
2. If "yes" to question 1: Should this check be inserted immediately after lines 82-85 (the existing error check)?
3. Is the value of the "tri" key in the surf parameter a single integer value?
4. Should a zero value also raise an error, or just purely negative values?

Depending on the answers to these questions, I might be willing to tackle this. (I am relatively new to making contributions to open source, but I have checked your readme file and guidelines for contributions.)

**Comment by ReinderVosDeWael** [comment]
Hi,

Firstly thanks for your interest in contributing. I just looked a bit more closely at this issue and I think the problem runs a bit deeper than what I had expected when I made this post. Somewhere in the writing of the SLM class the definition of the tri property has gotten lost. Unfortunately that may mean this wouldn't be an easy issue to get started with. 

I'll have to do a bit of a deeper dive to see whether I'm correct that its really no longer defined anywhere and if yes figure out what went wrong there. 

So to answer the above questions

1. Possibly, see above.
2. Probably this check would be inserted as part of a set function.
3. It's a n-by-3 array.
4. I'd have to double check - as we converted this code from MATLAB I'm not 100% sure whether we adjusted the tri variable to zero indexing or adjusted the functions to accept indexing starting from 1. I believe its the latter...
